src/math/trig.c

```diff
@@ -94,9 +97,198 @@ void cordic_sin_cos(int32_t th_rad_fxp, cordic_cfg type, int32_t *sign, int32_t
+			y = (y - ydshift) + sign * xshift;
+			z += sign * LUT[b_i];
+		}
+		b_cosvalue += b_cosvalue >> j;
+	}
+	if (z < 0)
+		z = -z;
+
+	return z;
+}
+
+/**
+ * \CORDIC-based approximation for inverse sine
+ * Arguments	: int32_t sinvalue
+ *		  int16_t numiters
+ *		  const int32_t LUT[CORDIC_30B_ITABLE_SIZE]
+ * Return Type	: int32_t
+ */
+int32_t is_scalar_cordic_asin(int32_t sinvalue, int16_t numiters,
+			      const int32_t LUT[CORDIC_30B_ITABLE_SIZE])
+{
+	int32_t b_sinvalue;
+	int32_t b_i;
+	int32_t i;
+	int32_t x = 0;
+	int32_t xdshift;
+	int32_t xshift;
+	int32_t y = 0;
+	int32_t ydshift;
+	int32_t yshift;
+	int32_t z = 0;
+	int32_t sign;
+	int16_t j;
+	int16_t k;
+
+	b_sinvalue = (sinvalue >> 1);
+	/* Initialize the variables for the cordic iteration
+	 * angles less than pi/4, we initialize (x,y) along the x-axis.
+	 * angles greater than or equal to pi/4, we initialize (x,y)
+	 * along the y-axis. This improves the accuracy of the algorithm
+	 * near the edge of the domain of convergence
+	 */
+	if (b_sinvalue > cord_arcsincos_q28fl) {
+		x = 0;
+		y = cord_arcsincos_q30fl;
+		z = PI_DIV2_Q3_29;
+	} else {
+		x = cord_arcsincos_q30fl;
+		y = 0;
+		z = 0;
+	}
+	b_sinvalue <<= 1;
+	/* DCORDIC(Double CORDIC) algorithm */
+	/* Double iterations method consists in the fact that unlike the classical */
+	/* CORDIC method,where the iteration step value changes EVERY time, i.e. on */
+	/* each iteration, in the double iteration method, the iteration step value */
+	/* is repeated twice and changes only through one iteration */
+	i = numiters - 1;
+	for (b_i = 0; b_i < i; b_i++) {
+		j = (b_i + 1) << 1;
+		if (j >= 31)
+			j = 31;
+
+		if (b_i < 31)
+			k = (int16_t)b_i;
+		else
+			k = 31;
+
+		xshift = x >> k;
+		xdshift = x >> j;
+		yshift = y >> k;
+		ydshift = y >> j;
+		/* Do nothing if x currently equals the target value. Allowed for
+		 * double rotations algorithms, as it is equivalent to rotating by
+		 * the same angle in opposite directions sequentially. Accounts for
+		 * the scaling effect of CORDIC Pseudo-rotations as well.
+		 */
+		if (y == b_sinvalue) {
+			x += xdshift;
+			y += ydshift;
+		} else {
+			sign = (((y >= b_sinvalue) && (x >= 0)) ||
+				((y < b_sinvalue) && (x < 0))) ? 1 : -1;
+			x = (x - xdshift) + sign * yshift;
+			y = (y - ydshift) - sign * xshift;
+			z -= sign * LUT[b_i];
+		}
+		b_sinvalue += b_sinvalue >> j;
+	}
+	if (z < 0)
+		z = -z;
+
+	return z;
+}
+
+/**
+ * approximated complex result
+ * Arguments	: int32_t sign
+ *		  int32_t b_yn
+ *		  int32_t xn
+ *		  enum    type
+ * Return Type	: struct
+ */
+struct cordic_cmpx cmpx_cexp(int32_t sign, int32_t b_yn, int32_t xn, cordic_cfg type,
+			     struct cordic_cmpx *cexp)
+{
+	cexp->re = sign * xn;
+	cexp->im = sign * b_yn;
+	/*convert Q2.30 to Q17.15*/
+	if (type == EN_16B_CORDIC_CEXP) {
+		cexp->re =  sat_int16(Q_SHIFT_RND((cexp->re), 30, 15));
+		cexp->im = sat_int16(Q_SHIFT_RND((cexp->im), 30, 15));
+	}
+	return *cexp;
```

Comment:
you dont need to return cexp since it's being passed in as an argument. Best to make this a void function.

Reply:
OK, your suggestion looks fine . I will take it. Thanks